DashKit/PodSDKProtocol/PodSDKLoggingShim.swift

```diff
@@ -0,0 +1,205 @@
+    weak var loggingShimDelegate: PodSDKLoggingShimDelegate?
+    
+
+    init(target: PodCommManagerProtocol) {
+        self.target = target
+    }
+    
+    var delegate: PodCommManagerDelegate? {
+        set {
+            target.delegate = newValue
+        }
+        get {
+            return target.delegate
+        }
+    }
+    
+    private func logSend(_ message: String, completion: ((Error?) -> Void)? = nil) {
+        loggingShimDelegate?.podSDKLoggingShim(self, didLogEventForDevice: deviceIdentifier, type: .send, message: message, completion: completion)
+    }
+    
+    private func logReceive(_ message: String, completion: ((Error?) -> Void)? = nil) {
+        loggingShimDelegate?.podSDKLoggingShim(self, didLogEventForDevice: deviceIdentifier, type: .receive, message: message, completion: completion)
+    }
+}
+
+extension PodSDKLoggingShim: PodCommManagerProtocol {
+    
+    func setLogger(logger: LoggingProtocol) {
+        target.setLogger(logger: logger)
+    }
+        
+    func enableAutoConnection(launchOptions: [AnyHashable : Any]?) {
+        logSend("enableAutoConnection(\(String(describing: launchOptions)))")
+        target.enableAutoConnection(launchOptions: launchOptions)
+    }
+    
+    func startPodActivation(lowReservoirAlert: LowReservoirAlert?, podExpirationAlert: PodExpirationAlert?, eventListener: @escaping (ActivationStatus<ActivationStep1Event>) -> ()) {
+        logSend("startPodActivation(\(String(describing: lowReservoirAlert)), \(String(describing: podExpirationAlert)))")
+        target.startPodActivation(lowReservoirAlert: lowReservoirAlert, podExpirationAlert: podExpirationAlert) { (event) in
+            self.logReceive("startPodActivation event: \(event)")
+            eventListener(event)
+        }
+    }
+    
+    func finishPodActivation(basalProgram: ProgramType, autoOffAlert: AutoOffAlert?, eventListener: @escaping (ActivationStatus<ActivationStep2Event>) -> ()) {
+        logSend("finishPodActivation(\(basalProgram), \(String(describing: autoOffAlert)))")
+        target.finishPodActivation(basalProgram: basalProgram, autoOffAlert: autoOffAlert) { (event) in
+            self.logReceive("finishPodActivation event: \(event)")
+            eventListener(event)
+        }
+    }
+    
+    func discardPod(completion: @escaping (PodCommResult<Bool>) -> ()) {
+        logSend("discardPod()")
+        target.discardPod { (result) in
+            self.logReceive("discardPod result: \(result)")
+            completion(result)
+        }
+    }
+    
+    func deactivatePod(completion: @escaping (PodCommResult<PodStatus>) -> ()) {
+        logSend("deactivatePod()")
+        target.deactivatePod { (result) in
+            self.logReceive("deactivatePod result: \(result)")
+            completion(result)
+        }
+    }
+    
+    func getPodStatus(userInitiated: Bool, completion: @escaping (PodCommResult<PodStatus>) -> ()) {
+        logSend("getPodStatus()")
```

Comment:
[nit] Add `userInitiated` to `logSend` for consistency with other functions here?